lib/srv/app/transport.go

```diff
@@ -280,3 +286,16 @@ func isRedirect(code int) bool {
 	}
 	return false
 }
+
+// websocketsRewriter rewrites websocket requests going through app access.
+//
+// Implements forward.ReqRewriter.
+type websocketsRewriter struct {
+	uri *url.URL
+}
+
+// Rewrite rewrites the websockets request.
+func (r *websocketsRewriter) Rewrite(req *http.Request) {
+	// Update host to the target app's host to make sure it's forwarded correctly.
+	req.URL.Host = r.uri.Host
+}
```

Comment:
This doesn't go through the round tripper?

Reply:
No, websockets use their own forwarder logic and a separate rewriter as well:

https://github.com/gravitational/oxy/blob/master/forward/fwd.go#L124-L128